unittests/ASM/FEX_bugs/MinMaxNaN.asm

```diff
@@ -0,0 +1,118 @@
+%ifdef CONFIG
+{
+  "RegData": {
+      "RAX": "0x00000000",
+      "RBX": "0x00000000",
+      "RCX": "0x00000000",
+      "RSI": "0x00000000"
+  }
+}
+%endif
+
+%macro single_case 4
+  ; Load sources
+  mov eax, %2
+  mov ebx, %3
+  movd xmm0, eax
+  movd xmm1, ebx
+
+  ; Calculate scalar min/max
+  %1ss xmm0, xmm1
+
+  ; Check result
+  movd ecx, xmm0
+  cmp ecx, %4
+  jne fexi_fexi_im_so_broken
+  mov ecx, 0
+
+  ; Now try the SSE vector
+  %1ps xmm0, xmm1
+  movd ecx, xmm0
+  cmp ecx, %4
+  jne fexi_fexi_im_so_broken
+  mov ecx, 0
+
+  ; And the AVX-128 version
+  v%1ps xmm2, xmm0, xmm1
+  movd ecx, xmm2
+  cmp ecx, %4
+  jne fexi_fexi_im_so_broken
+  mov ecx, 0
+
+  ; And the AVX-256 version
+  v%1ps ymm2, ymm0, ymm1
+  movd ecx, xmm2
+  cmp ecx, %4
+  jne fexi_fexi_im_so_broken
+
+%endmacro
+
+%macro min 3
+  single_case min, %1, %2, %3
+%endmacro
+
+%macro max 3
+  single_case max, %1, %2, %3
+%endmacro
+
+zero equ 0x00000000
+negzero equ 0x80000000
+qnan equ 0x7fc00000
+snan equ 0x7fc00000
```

Comment:
Is snan and qnan expected to be the same here?

Reply:
doh. No, of course not, copypaste fail

Reply:
fixed, thanks!